src/main/scala/com/lightbend/kafkalagexporter/GraphiteEndpointSink.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package com.lightbend.kafkalagexporter
+
+import com.lightbend.kafkalagexporter.MetricsSink._
+import com.lightbend.kafkalagexporter.EndpointSink.ClusterGlobalLabels
+import java.net.Socket
+import java.io.PrintWriter
+import scala.util.{Try, Success, Failure}
+
+import scala.util.Try
+
+object GraphiteEndpointSink {
+
+  def apply(metricWhitelist: List[String], clusterGlobalLabels: ClusterGlobalLabels,
+            graphiteConfig: Option[GraphiteConfig]): MetricsSink = {
+    Try(new GraphiteEndpointSink(metricWhitelist, clusterGlobalLabels, graphiteConfig))
+      .fold(t => throw new Exception("Could not create Prometheus Endpoint", t), sink => sink)
+  }
+}
+
+class GraphiteEndpointSink private(metricWhitelist: List[String], clusterGlobalLabels: ClusterGlobalLabels,
+                                      graphiteConfig: Option[GraphiteConfig]) extends EndpointSink(clusterGlobalLabels) {
+  def graphitePush(graphiteConfig: GraphiteConfig, metricName: String, metricValue: Double): Unit = {
+    Try(new Socket(graphiteConfig.host, graphiteConfig.port)) match {
+      case Success(socket) =>
+        Try(new PrintWriter(socket.getOutputStream)) match {
+          case Success(writer) =>
+            writer.print(s"${graphiteConfig.prefix.getOrElse("")}${metricName} ${metricValue} ${System.currentTimeMillis / 1000}\n")
+            writer.close
+            socket.close
+          case Failure(_) =>
+            socket.close
+        }
+      case Failure(_) => {
+      }
+    }
+  }
+
+  def metricNameToGraphiteMetricName(metricValue: MetricValue): String = {
```

Comment:
I'm not sure I understand what's happening here.  The metrics are already snakecase.  Can you add a scaladoc comment with an example of the conversion?

Reply:
added @example

Reply:
I see.